SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialStatics.cpp

```diff
@@ -101,62 +89,20 @@ bool USpatialStatics::IsActorGroupOwnerForActor(const AActor* Actor)
 
 bool USpatialStatics::IsActorGroupOwnerForClass(const UObject* WorldContextObject, const TSubclassOf<AActor> ActorClass)
 {
-	if (SpatialActorGroupManager* ActorGroupManager = GetActorGroupManager(WorldContextObject))
-	{
-		const FName ClassWorkerType = ActorGroupManager->GetWorkerTypeForClass(ActorClass);
-		const FName CurrentWorkerType = GetCurrentWorkerType(WorldContextObject);
-		return ClassWorkerType == CurrentWorkerType;
-	}
-
-	if (const UWorld* World = WorldContextObject->GetWorld())
-	{
-		return World->GetNetMode() != NM_Client;
-	}
-
-	return false;
-}
-
-bool USpatialStatics::IsActorGroupOwner(const UObject* WorldContextObject, const FName ActorGroup)
-{
-	if (SpatialActorGroupManager* ActorGroupManager = GetActorGroupManager(WorldContextObject))
+	const UWorld* World = WorldContextObject->GetWorld();
+	if (World == nullptr)
 	{
-		const FName ActorGroupWorkerType = ActorGroupManager->GetWorkerTypeForActorGroup(ActorGroup);
-		const FName CurrentWorkerType = GetCurrentWorkerType(WorldContextObject);
-		return ActorGroupWorkerType == CurrentWorkerType;
+		return false;
 	}
 
-	if (const UWorld* World = WorldContextObject->GetWorld())
+	if (const USpatialNetDriver* SpatialNetDriver = Cast<USpatialNetDriver>(World->GetNetDriver()))
 	{
-		return World->GetNetMode() != NM_Client;
-	}
-
-	return false;
-}
-
-FName USpatialStatics::GetActorGroupForActor(const AActor* Actor)
-{
-	if (SpatialActorGroupManager* ActorGroupManager = GetActorGroupManager(Actor))
-	{
-		const AActor* EffectiveActor = Actor;
-		while (EffectiveActor->bUseNetOwnerActorGroup && EffectiveActor->GetOwner() != nullptr)
+		if (const ULayeredLBStrategy* LBStrategy = Cast<ULayeredLBStrategy>(SpatialNetDriver->LoadBalanceStrategy))
```

Comment:
There's a case where we have SpatialNetDriver by no LBStrategy and calling could have authority fails without error. If there's no LBStrategy then a) CouldHaveAuthority should effectively be true since we're using single-server spatial and b) we could arguably print a warning since calling this in a single-server context doesn't make sense (although maybe we don't since toggling multi worker would require adding the conditional check in game logic)

Reply:
You're right, although the right thing to do here is just have the default return be true I think.  (And yes, the way the Scavengers code is written, they want to be able to call this to figure out whether they're doing Offloading at all, which they should really do with a different method. As mentioned earlier, I want all this to go away as soon as we get them to change that call, but I have to support their use case for now.